### DISCUSSION TITLE
Changes to node selection for basic k3s master upgrade

### DISCUSSION THREAD

**Discussion by AntonOfTheWoods** [rancher/system-upgrade-controller#175] [Q&A]
I started using the upgrade controller a while back and everything worked fine. I have a few single node clusters and one 5-node (production) cluster (all nodes are identical masters, using built-in etcd). I completely reinstalled all but one single node cluster, and now the upgrade controller no longer works on anything that has been installed recently. I have now spent many hours trying to get upgrade working again to no avail. I continue to be able to upgrade my non-reinstalled node by simply updating version to the latest version, eg. `version: v1.22.5+k3s1`, but this has no longer has any effect on any of the other clusters. I have tried various combinations of the `rancher/system-upgrade-controller` image but nothing works. I have been focussing on trying to get one of the other single node clusters to upgrade and have had zero success. A diff of the yamls of the deployments and plans gives differences only with creationTimestamp, generation, resourceVersion and uid. On the "old" cluster every update starts immediately, on all of the others nothing happens at all, ever.

Is there something obvious I'm missing?

**Comment by dweomer** [comment]
Likely the Plan's NodeSelector needs to be updated as `node-role.kubernetes.io/master` was deprecated some time after I began work on SUC. i.e. this needs updating: https://github.com/rancher/system-upgrade-controller/blob/master/manifests/system-upgrade-controller.yaml#L60
